src/core/simulation_sequence.jl

```diff
@@ -1,51 +1,193 @@
+"""
+    _calculate_interval_inner_counts(order::Dict{Int64,String},
+                                          intervals::Dict{String,<:Dates.TimePeriod},
+                                          step_resolution::Dates.TimePeriod)
+
+Calculates how many times a stage is executed for every interval of the previous stage
+"""
+function _calculate_interval_inner_counts(
+    order::Dict{Int64,String},
+    intervals::Dict{String,<:Dates.TimePeriod},
+    step_resolution::Dates.TimePeriod,
+)
+    reverse_order = sort(collect(keys(order)), rev = true)
+    interval_run_counts = Dict{Int64,Int64}()
+    for k in reverse_order[1:end-1]
+        stage_name = order[k]
+        previous_stage_name = order[k-1]
+        stage_interval = intervals[stage_name]
+        previous_stage_interval = intervals[previous_stage_name]
+        try
+            interval_run_counts[k] = previous_stage_interval / stage_interval
+        catch e
+            if isa(e, InexactError)
+                throw(IS.ConflictingInputsError("The interval configuration provided results in a fractional number of executions of stage $stage_name"))
+            end
+            throw(e)
+        end
+        @debug "Stage $k is executed $(interval_run_counts[k]) time within each interval of Stage $(k-1)"
+    end
+    stage_name = order[1]
+    stage_interval = intervals[stage_name]
+    interval_run_counts[1] = 1
+    return interval_run_counts
+end
+
+""" Function calculates the total number of stage executions in the simulation and allocates the appropiate vector"""
+function _allocate_execution_order(interval_run_counts::Dict{Int64,Int64})
+    total_size_of_vector = 0
+    for k in sort(collect(keys(interval_run_counts)))
+        mult = 1
+        for i in 1:k
+            mult = mult * interval_run_counts[i]
+        end
+        total_size_of_vector += mult
+    end
+    return -1 * ones(Int64, total_size_of_vector)
+end
+
+function _fill_execution_order(
+    execution_order::Vector{Int64},
+    interval_run_counts::Dict{Int64,Int64},
+)
+    function _fill_stage(
+        execution_order::Vector{Int64},
+        index::Int64,
+        stage::Int64,
+        interval_run_counts::Dict{Int64,Int64},
+        last_stage::Int64,
+    )
+        if stage < last_stage
+            next_stage = stage + 1
+            for i in 1:interval_run_counts[next_stage]
+                index = _fill_stage(
+                    execution_order,
+                    index,
+                    next_stage,
+                    interval_run_counts,
+                    last_stage,
+                )
+            end
+        end
+        execution_order[index] = stage
+        index -= 1
+    end
+
+    index = length(execution_order)
+    stages = sort!(collect(keys(interval_run_counts)))
+    last_stage = stages[end]
+    _fill_stage(execution_order, index, stages[1], interval_run_counts, last_stage)
+end
+
+function _get_execution_order_vector(
```

Comment:
Suggest it's worth having a unit test for this function.

Reply:
there are 2 tests already for the output of the function.